cmd/observer/node_builder/observer_builder.go

```diff
@@ -0,0 +1,552 @@
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/consensus/hotstuff/committees"
+	"github.com/onflow/flow-go/consensus/hotstuff/notifications/pubsub"
+	hotsignature "github.com/onflow/flow-go/consensus/hotstuff/signature"
+	"github.com/onflow/flow-go/consensus/hotstuff/verification"
+	recovery "github.com/onflow/flow-go/consensus/recovery/protocol"
+	"github.com/onflow/flow-go/engine/access/ingestion"
+	"github.com/onflow/flow-go/engine/access/rpc"
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/engine/common/follower"
+	followereng "github.com/onflow/flow-go/engine/common/follower"
+	"github.com/onflow/flow-go/engine/common/requester"
+	synceng "github.com/onflow/flow-go/engine/common/synchronization"
+	"github.com/onflow/flow-go/model/encodable"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/flow/filter"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/buffer"
+	finalizer "github.com/onflow/flow-go/module/finalizer/consensus"
+	"github.com/onflow/flow-go/module/id"
+	"github.com/onflow/flow-go/module/mempool/stdmap"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/module/synchronization"
+	"github.com/onflow/flow-go/network"
+	netcache "github.com/onflow/flow-go/network/cache"
+	cborcodec "github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/validator"
+	"github.com/onflow/flow-go/state/protocol"
+	badgerState "github.com/onflow/flow-go/state/protocol/badger"
+	"github.com/onflow/flow-go/state/protocol/blocktimer"
+	storage "github.com/onflow/flow-go/storage/badger"
+)
+
+// ObserverBuilder extends cmd.NodeBuilder and declares additional functions needed to bootstrap an Access node
+// These functions are shared by staked and observer node builders.
+// The Staked network allows the staked nodes to communicate among themselves, while the public network allows the
+// observers and a staked Access node to communicate.
+//
+//                                 public network                           staked network
+//  +------------------------+
+//  | observer Node 1        |<--------------------------|
+//  +------------------------+                           v
+//  +------------------------+                         +--------------------+                 +------------------------+
+//  | observer Node 2        |<----------------------->| Staked Access Node |<--------------->| All other staked Nodes |
+//  +------------------------+                         +--------------------+                 +------------------------+
+//  +------------------------+                           ^
+//  | observer Node 3        |<--------------------------|
+//  +------------------------+
+
+type ObserverBuilder interface {
+	cmd.NodeBuilder
+
+	// IsStaked returns True if this is a staked Access Node, False otherwise
+	IsStaked() bool
```

Comment:
observer config does not need this

Reply:
Thank you for the review. Fixed: https://github.com/GetElastech/flow-go/commit/1c6779722a115f15ad4da440e24eba0a5d451139
I still kept it as deprecated, so that it goes through tests and does not break boostrap code elsewhere.